app/src/main/kotlin/com/wire/android/media/CallRinger.kt

```diff
@@ -35,12 +51,39 @@ class CallRinger @Inject constructor(private val context: Context) {
 
     fun ring(resource: Int, isLooping: Boolean = true) {
         stop()
+        vibrateIfNeeded()
         createMediaPlayer(resource, isLooping)
+        appLogger.i("Starting ringing");
         mediaPlayer?.start()
     }
 
+
+    private fun vibrateIfNeeded() {
+        vibrator?.let {
+            if (!it.hasVibrator()) {
+                appLogger.i("Device does not support vibration");
+                return
+            }
+            val audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager?
+            val ringerMode = audioManager?.ringerMode
+            if (ringerMode == AudioManager.RINGER_MODE_VIBRATE) {
+                appLogger.i("Starting vibration");
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                    it.vibrate(VibrationEffect.createWaveform(VIBRATE_PATTERN, VibrationEffect.EFFECT_DOUBLE_CLICK))
+                } else {
+                    @Suppress("DEPRECATION")
+                    it.vibrate(VIBRATE_PATTERN, VibrationEffect.EFFECT_DOUBLE_CLICK)
+                }
+            } else {
+                appLogger.i("Skipping vibration");
+            }
+        }
+    }
+
     fun stop() {
+        appLogger.i("Stopping ringing");
         try {
+            vibrator?.cancel()
```

Comment:
Is there any option to `release`/`destroy` the vibrator after it is being used? Not here necessarily but just so we don't have it leaked? 

Just a question, I have no idea what `cancel` actually does